zingolib/src/lightclient/send.rs

```diff
@@ -248,6 +250,17 @@ pub mod send_with_proposal {
 
             let step = proposal.steps().first();
 
+            fn usk_to_tkey(
+                unified_spend_key: &UnifiedSpendingKey,
+                t_metadata: &TransparentAddressMetadata,
+            ) -> SecretKey {
+                hdwallet::ExtendedPrivKey::deserialize(&unified_spend_key.transparent().to_bytes())
+                    .expect("This a hack to do a type conversion, and will not fail")
+                    .derive_private_key(t_metadata.address_index().into())
+                    .expect("check this expect, maybe this fn should be fallible")
```

Comment:
This expect is not going to remain. The rest of this PR is ready for review, except I have merge conflicts to resolve.